source/agora/common/Config.d

```diff
@@ -96,6 +96,10 @@ public struct NodeConfig
     static assert(!hasUnsharedAliasing!(typeof(this)),
         "Type must be shareable accross threads");
 
+    /// If set, a hexdump serialized representation of the genesis block to use
+    /// in place of the built-in genesis block as defined by CoinNet
+    public string genesis_block;
```

Comment:
No default ?

Reply:
If there's a default here then it will always override the initialization that's set in `Genesis.d`:

```
/// Contains a pointer to the genesis block.
version (unittest)
    private immutable(Block)* gen_block = &UnitTestGenesisBlock;
else
    private immutable(Block)* gen_block = &CoinNetGenesisBlock;
```

On the other hand, if we only provide here in the Config struct, then we could remove all the definitions in `Genesis.d`. But I'm not sure if that's the right approach. 🤔

Reply:
I think I can do new blockchain(fork?) easily if I just change the config file.

Reply:
Forking refers to splitting the network from an origin point. If networks differ in their genesis block then they do not fork, because they don't have a common ancestor (the genesis block)